packages/cardano-services/src/Program/services/ogmios.ts

```diff
@@ -140,16 +141,19 @@ export const ogmiosCardanoNodeWithDiscovery = async (
   });
 };
 
-export const getOgmiosCardanoNode = async (
-  dnsResolver: DnsResolver,
-  logger: Logger,
-  options?: CommonProgramOptions
-): Promise<OgmiosCardanoNode> => {
-  if (options?.ogmiosSrvServiceName)
-    return ogmiosCardanoNodeWithDiscovery(dnsResolver, logger, options.ogmiosSrvServiceName);
-  if (options?.ogmiosUrl) return new OgmiosCardanoNode(urlToConnectionConfig(options.ogmiosUrl), logger);
-  throw new MissingProgramOption(ServiceNames.TxSubmit, [
-    CommonOptionDescriptions.OgmiosUrl,
-    CommonOptionDescriptions.OgmiosSrvServiceName
-  ]);
-};
+export const getOgmiosCardanoNode = memoize(
```

Comment:
I think it never will memoize one and same instance of `ogmiosCardanoNode` since we use `await getOgmiosCardanoNode()` with different arguments for both services [StakePool](https://github.com/input-output-hk/cardano-js-sdk/pull/386/files#diff-cfc78f77666af1469e2e670c258fecabed5a923995563490c52370f7a6b4888cR86) and [NetworkInfo](https://github.com/input-output-hk/cardano-js-sdk/pull/386/files#diff-cfc78f77666af1469e2e670c258fecabed5a923995563490c52370f7a6b4888cR112) as it caches the callbacks by given a set of arguments :thinking_face:? @mkazlauskas  @jvieiro

Reply:
By default `memoize` only considers the 1st argument. If it doesn't match then you can add a 2nd `resolver` argument to memoize as `() => true`.

It should be refactored to **not** take in `serviceName` because the goal of memoize here is to use a single instance of OgmiosCardanoNode for **multiple** different services.

Reply:
After a fast look into it, I see `serviceName` is passed only to be used by `throw new MissingProgramOption` to expose which is the exact service with missing program options in the error. This could be changed if we don't care about it and throw a different error (even if we never go in that case because have defaulted to Ogmios URL always)